### ISSUE TITLE
A11y_.NETCore_ASP.NETCore_FetchData_ScreenReader: Narrator is announcing the column 2 & 3 as by letter TEMP C which is not conveying the meaning as Temperature in Celsius & Temperature in Fahrenheit.

### ISSUE THREAD

**Issue by PoojaNamde** [dotnet/aspnetcore#49793]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Github Tags:
#A11yMAS; #A11ySev3; #A11yTCS; #DesktopWeb; #BM_.NETCore_Web_Jul23; #.NET Core; #Win11; #ChromiumEdge; #ScreenReader; #NVDA; #Narrator; #Chrome; #WCAG1.3.1;

### Environment Details:
App name: .NET Core
URL: http://localhost:5268/
Chromium Edge: Version 116.0.1938.29 (Official build) dev (64-bit)
Chrome: Version 115.0.5790.171 (Official Build) (64-bit)
Screen Reader: Narrator, NVDA 2022.4

### Repro Steps:

1. Open command prompt and enter command "mkdir foldername" e.g. "mkdir BlazorWasm".
2. Move to the created directory using command "cd foldername".
3. Run command "dotnet new blazorwasm" and hit ENTER key.
4. Run command "dotnet run".
5. Copy the First URL link from Command prompt and paste it in URL section of Browser and hit ENTER key.
6. TAB to "Fetch Data" and hit ENTER key.
7. Start Narrator screen reader and navigate to the table.

### Actual Result:
Narrator is announcing the column 2 & 3 as by letter TEMP C and TEMP F which is not conveying the meaning as Temperature in Celsius & Temperature in Fahrenheit.

### Expected Result:
Narrator should announce the column TEMP C and TEMP F as "Temperature in Celsius and Temperature in Fahrenheit".

### User Impact:
Screen reader users will not get correct information about the column name and will get confuse here.

### Attachment:
[49793_A11y_.NETCore_ASP.NETCore_FetchData_NVDA.webm](https://github.com/dotnet/aspnetcore/assets/90901519/fdb83122-47d2-4153-9180-bc890dc8ce4a)
[49793_A11y_.NETCore_ASP.NETCore_FetchData_Narrator.webm](https://github.com/dotnet/aspnetcore/assets/90901519/0a8986f0-4d53-4d3a-b2c9-9e4458966438)

**Comment by merriemcgaw** [comment]
Note that this is a Sev3 issues, so there is no specific SLA involved. The feature team can decide when/if to fix this issue.

**Comment by PoojaNamde** [comment]
#Regressed:09-12-23;

As per above comment there is no action pending at our end.

**Comment by javiercn** [comment]
The change is to use aria-label to define alternative text for the elements.

```
   <tr>
     <th>Date</th>
     <th aria-label="Temperature in Celsius">Temp. (C)</th>
     <th aria-label="Temperature in Farenheit">Temp. (F)</th>
     <th>Summary</th></tr>
```

**Comment by PoojaNamde** [comment]
@javiercn  Could you please share latest zip file to verify this issue.
